src/core/public/chrome/ui/header/recent_items.tsx

```diff
@@ -94,46 +147,132 @@ export const RecentItems = ({
     </EuiToolTip>
   );
 
+  useEffect(() => {
+    const savedObjects = recentlyAccessedItems
+      .filter((item) => item.meta?.type)
+      .map((item) => ({
+        type: item.meta?.type || '',
+        id: item.id,
+      }));
+
+    if (savedObjects.length) {
+      bulkGetDetail(savedObjects, http).then((res) => {
+        const formatDetailedSavedObjects = res.map((obj) => {
+          const recentAccessItem = recentlyAccessedItems.find(
+            (item) => item.id === obj.id
+          ) as ChromeRecentlyAccessedHistoryItem;
+
+          const findWorkspace = workspaceList.find(
+            (workspace) => workspace.id === recentAccessItem.workspaceId
+          );
+
+          return {
+            ...recentAccessItem,
+            ...obj,
+            ...recentAccessItem.meta,
+            updatedAt: moment(obj?.updated_at).valueOf(),
+            workspaceName: findWorkspace?.name,
+          };
+        });
+        setDetailedSavedObjects(formatDetailedSavedObjects);
+      });
+    }
+  }, [recentlyAccessedItems, http, workspaceList]);
+
+  const selectedRecentsItems = useMemo(() => {
+    return detailedSavedObjects.slice(0, Number(recentsRadioIdSelected.split('recent_')[1]));
```

Comment:
Nit: We could add an attribute in recentsRadios to store actual value and find instead of doing string split next time.

Reply:
Thank you, updated.